documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc

```diff
@@ -140,6 +91,12 @@ The default bootloader configuration on CM4 is designed to support bringup and d
 
 N.B. The Compute Module 4 ROM never runs `recovery.bin` from SD/EMMC and the `rpi-eeprom-update` service is not enabled by default. This is necessary because the EMMC is not removable and an invalid `recovery.bin` file would prevent the system from booting. This can be overridden and used with `self-update` mode where the bootloader can be updated from USB MSD or Network boot. However, `self-update` mode is not an atomic update and therefore not safe in the event of a power failure whilst the EEPROM was being updated.
 
+==== Flashing NVMe / other storage devices.
+The new Linux based https://github.com/raspberrypi/usbboot/blob/master/mass-storage-gadget/README.md[mass-storage gadget] supports flashing of NVMe, EMMC and USB block devices.
```

Comment:
Only a minor nitpick, but "Linux based" should probably be "Linux-based" ?

Reply:
*shrug* ok